inorbit_edge/robot.py

```diff
@@ -536,63 +558,145 @@ def _stop_cameras_streaming(self):
             for s in self.camera_streamers.values():
                 s.stop()
 
-    def publish_map(
-        self,
-        file,
-        map_id="map",
-        frame_id="map",
-        x=0,
-        y=0,
-        resolution=0.05,
-        ts=None,
-        is_update=False,
-    ):
-        """
-        Sends the information about a map and optionally its contents. The map message
-        data is passed as a separate variable to indicate that the map data needs to
-        be actually sent. An is_update flag can be provided to inform in the message
-        that this is an update message, aimed to correct a previous one, currently
-        used for frame_id/map_id updates.
-        """
+    def _read_map_file(self, file: str) -> Optional[bytes]:
+        """Read a map file and return its bytes. If loading fails return None and log
+        an error."""
+
         try:
             # Open the image file
             img = Image.open(file)
-            # Check if image file is a png image
-            if img.format != "PNG":
-                self.logger.error(f"{file} is not a PNG image")
-                return None
             # Verify opens and reads the entire image file
             img.verify()
         except IOError:
             self.logger.error(f"{file} is not accessible.")
-            return None
+            return (None, None, None)
 
         # img.verify() closes the file, reload it now that its validated
         img = Image.open(file)
 
         # Create a BytesIO object
         img_byte_arr = io.BytesIO()
         img.save(img_byte_arr, format="PNG")
-        img_byte_arr = img_byte_arr.getvalue()
 
+        return (img_byte_arr.getvalue(), img.width, img.height)
+
+    def _publish_map_bytes(
```

Comment:
What do you think about naming this `_publish_map` to make it clearer ? Any data is bytes